docs/local_dev_environment.md

```diff
@@ -0,0 +1,212 @@
+
+```console
+python3 -m venv py3-venv
+. py3-venv/bin/activate
+```
+
+## Set up local Postgresql DB for testing
+
+You can use a local postgresql for testing purposes.
+
+### Set up local Postgresql DB on WSL
+
+You can use `bash` to install postgres:
+
+```console
+sudo apt install postgresql-client-common
+sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
+wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+sudo apt-get update
+sudo apt-get install postgresql-12
+```
+
+Make sure the cluster is started:
+
+```console
+sudo pg_ctlcluster 12 main start
+```
+
+### Set up local Postgresql DB on Mac
+
+If you're on mac, you can install postgres using brew:
+
+```console
+brew install postgres
+```
+
+### Set up DB and users for testing
+
+You'll need to connect to the postgresql and add test users and databases.
+
+#### Connect to Postgresql on WSL
+
+Connect to the local postgresql server
+
+```console
+sudo -i -u postgres
+psql
+```
+
+#### Connect to Postgresql on Mac
+
+If you're on a mac, use the following to connect to postgres:
+
+```console
+brew services start postgres
+psql postgres
+```
+
+#### Helpful psql commands
+It may be helpful to understand some psql commands too:
+
+```console
+\conninfo # check connection info
+\l # list databases
+\d # list tables in database
+\c # list short connection info
+\c postgres # connect to a database named postgres
+\q # quit
+```
+
+#### Set up users in psql
+
+Initialize a user within the psql console:
+
+```console
+CREATE USER postgres WITH PASSWORD 'test';
+ALTER USER postgres WITH PASSWORD 'test';
+\du
+```
+
+## Installation
+
+### Install for General Usage
+
+To install sheepdog for use with other Gen3 services, running these commands is sufficient.
+
+```console
+python3 setup.py build
+python3 setup.py install
+```
+
+### Install for Development
+
+Check the [dev-requirements.txt](https://github.com/uc-cdis/sheepdog/blob/master/dev-requirements.txt) and make sure the requirements includes the following dependencies:
+
+```
+coverage==5.3.0
+codacy-coverage==1.3.11
+```
+
+Install the development requirements with the following commands:
+
+```console
+python3 -m pip install -r dev-requirements.txt
+python3 setup.py develop
+```
+
+Install requirements with pip.
+
+> One of the requirements `dictionaryutils==3.4.0` may not be available through pip, so you can try with a lower available version.  In this case, update [requirements.txt](https://github.com/uc-cdis/sheepdog/blob/master/requirements.txt) to use `dictionaryutils==3.2.0`.  You may also need to update the requirement for `gen3authz==0.2.1` to `gen3authz==0.4.0`.
+
+```console
+python3 -m pip install -r requirements.txt
+```
+
+### Validate Installation
+
+You can try to run the following to confirm installation:
+
+```console
+python3 sample_usage.py
```

Comment:
i'm not sure we should recommend this `sample_usage.py` instead of the `run.py` we already have. what do you think?

Reply:
agreed, no need for duplicate work. Let me add this as a comment to the original PR